diff --git a/bpf/tpinjector/maps/iter_listening_ports.h b/bpf/tpinjector/maps/iter_listening_ports.h
new file mode 100644
index 000000000..8706f3d09
--- /dev/null
+++ b/bpf/tpinjector/maps/iter_listening_ports.h
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <bpfcore/vmlinux.h>
+#include <bpfcore/bpf_helpers.h>
+
+#include <common/map_sizing.h>
+#include <common/sock_port_ns.h>
+
+// A duplicate map for listening ports, to ensure the tcp_iter and sock_iter
+// can operate independently.
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __type(key, struct sock_port_ns);
+    __type(value, bool);
+    __uint(max_entries, MAX_CONCURRENT_REQUESTS);
+} iter_listening_ports SEC(".maps");
diff --git a/bpf/tpinjector/sock_iter.c b/bpf/tpinjector/sock_iter.c
index 74cf07c4e..558559796 100644
--- a/bpf/tpinjector/sock_iter.c
+++ b/bpf/tpinjector/sock_iter.c
@@ -7,12 +7,15 @@
 #include <bpfcore/bpf_endian.h>
 
 #include <common/protocol_defs.h>
+#include <common/sock_port_ns.h>
 
 #include <logger/bpf_dbg.h>
 
 #include <maps/sock_dir.h>
 #include <maps/tracked_sock_cookies.h>
 
+#include <tpinjector/maps/iter_listening_ports.h>
+
 char __license[] SEC("license") = "Dual MIT/GPL";
 
 // max IPv6+port: "[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]:65535" = 48 chars
@@ -83,6 +86,31 @@ static __always_inline void format_sock_addrs(struct sock_common *skc,
     }
 }
 
+// First pass, we record every local listening port into iter_listening_ports.
+SEC("iter/tcp")
+int obi_sk_iter_tcp_listen(struct bpf_iter__tcp *ctx) {
+    struct sock_common *skc = ctx->sk_common;
+
+    if (!skc) {
+        return 0;
+    }
+
+    u8 skc_state = BPF_CORE_READ(skc, skc_state);
+
+    if (skc_state != TCP_LISTEN) {
+        return 0;
+    }
+
+    struct sock_port_ns pn = sock_port_ns_from_skc(skc);
+    bpf_map_update_elem(&iter_listening_ports, &pn, &(bool){true}, BPF_ANY);
+
+    struct seq_file *seq = ctx->meta->seq;
+    BPF_SEQ_PRINTF(seq, "Listening port=%u netns=%u\n", pn.port, pn.netns);
+
+    return 0;
+}
+
+// Second pass, track only active client connections in sock_dir.
 SEC("iter/tcp")
 int obi_sk_iter_tcp(struct bpf_iter__tcp *ctx) {
     struct sock_common *skc = ctx->sk_common;
@@ -91,6 +119,19 @@ int obi_sk_iter_tcp(struct bpf_iter__tcp *ctx) {
         return 0;
     }
 
+    // don't look at sockets that aren't established
+    u8 skc_state = BPF_CORE_READ(skc, skc_state);
+    if (skc_state != TCP_ESTABLISHED) {
+        return 0;
+    }
+
+    // skip passive sockets, their local port is a listening port we
+    // found in the prior pass
+    struct sock_port_ns pn = sock_port_ns_from_skc(skc);
+    if (bpf_map_lookup_elem(&iter_listening_ports, &pn) != 0) {
+        return 0;
+    }
+
     const u64 cookie = bpf_get_socket_cookie(skc);
 
     char src_buf[k_addr_buf_len] = {};
@@ -102,8 +143,6 @@ int obi_sk_iter_tcp(struct bpf_iter__tcp *ctx) {
 
     BPF_SEQ_PRINTF(seq, "Tracking socket cookie=%llu src=%s dst=%s\n", cookie, src_buf, dst_buf);
 
-    bpf_d_printk("Tracking socket cookie=%llu src=%s dst=%s", cookie, src_buf, dst_buf);
-
     if (bpf_map_update_elem(&sock_dir, &cookie, skc, BPF_NOEXIST) != 0) {
         bpf_dbg_printk("Failed to track sock cookie=%llu", cookie);
     } else {
diff --git a/pkg/internal/ebpf/tpinjector/tpinjector.go b/pkg/internal/ebpf/tpinjector/tpinjector.go
index 5a677e47f..2ac375f78 100644
--- a/pkg/internal/ebpf/tpinjector/tpinjector.go
+++ b/pkg/internal/ebpf/tpinjector/tpinjector.go
@@ -207,7 +207,13 @@ func (p *Tracer) Iters() []*ebpfcommon.Iter {
 		return p.iters
 	}
 
-	p.iters = []*ebpfcommon.Iter{{Program: p.bpfIterObjects.ObiSkIterTcp}}
+	// The ordering matters, we don't want to add passive listeners to
+	// the map, so we first find the listening ports and then we discard
+	// the established with those listening ports.
+	p.iters = []*ebpfcommon.Iter{
+		{Program: p.bpfIterObjects.ObiSkIterTcpListen},
+		{Program: p.bpfIterObjects.ObiSkIterTcp},
+	}
 
 	return p.iters
 }
