diff --git a/metrics/target.go b/metrics/target.go
index 0243c87..5a2d40f 100644
--- a/metrics/target.go
+++ b/metrics/target.go
@@ -6,8 +6,7 @@ import (
 	"maps"
 	"net"
 	"net/url"
-	"strconv"
 	"strings"
 	"time"
 
 	"github.com/coroot/coroot-cluster-agent/common"
@@ -116,7 +115,6 @@ func (t *Target) StartExporter(reg *prometheus.Registry, credentials Credentials
 		userPass := url.UserPassword(credentials.Username, credentials.Password)
 		query := url.Values{}
 		query.Set("connect_timeout", "1")
-		query.Set("statement_timeout", strconv.Itoa(int(collectTimeout.Milliseconds())))
 		sslmode := t.Params["sslmode"]
 		pqTLSName := ""
 		if tlsCreds.CA != "" || (tlsCreds.Cert != "" && tlsCreds.Key != "") {
