diff --git a/metrics/target.go b/metrics/target.go
index d392c5a..3e39831 100644
--- a/metrics/target.go
+++ b/metrics/target.go
@@ -6,7 +6,6 @@ import (
 	"maps"
 	"net"
 	"net/url"
-	"strconv"
 	"strings"
 	"sync"
 	"time"
@@ -133,7 +132,6 @@ func (t *Target) StartExporter(reg *prometheus.Registry, credentials Credentials
 		userPass := url.UserPassword(credentials.Username, credentials.Password)
 		query := url.Values{}
 		query.Set("connect_timeout", "1")
-		query.Set("statement_timeout", strconv.Itoa(int(collectTimeout.Milliseconds())))
 		sslmode := t.Params["sslmode"]
 		pqTLSName := ""
 		if tlsCreds.CA != "" || (tlsCreds.Cert != "" && tlsCreds.Key != "") {
